ugvc/mrd/featuremap_consensus_utils.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from os.path import basename
+from os.path import join as pjoin
+
+import numpy as np
+import pysam
+
+
+def write_a_pileup_record(
+    record_dict: dict, rec_id: str, out_fh: pysam.VariantFile, header: pysam.VariantHeader, min_qual: int
+):
+    info_fields = [
+        "X_QUAL",
+        "X_RN",
+    ]
+
+    format_fields = ["GT", "AD", "DP", "VAF"]
+
+    rec = header.new_record()
+    rec.chrom = rec_id.split("-")[0]
+    rec.pos = int(rec_id.split("-")[1])
+    rec.ref = rec_id.split("-")[2]
+    rec.alts = [rec_id.split("-")[3]]
+    rec.id = "."
+    rec.qual = np.mean(record_dict["X_QUAL"])
+    record_dict["X_RN"] = "|".join(record_dict["X_RN"])
+    curr_filter = "PASS" if rec.qual >= min_qual else "FAIL"
+    record_dict["VAF"] = record_dict["FILTERED_COUNT"] / record_dict["DP"]
+    rec.filter.add(curr_filter)
+
+    for info_field in info_fields:
+        rec.info[info_field] = record_dict[info_field]
+
+    record_dict["GT"] = (0, 1)
+    record_dict["AD"] = [record_dict["FILTERED_COUNT"], record_dict["DP"] - record_dict["FILTERED_COUNT"]]
+    for format_field in format_fields:
+        rec.samples["SAMPLE"][format_field] = record_dict[format_field]
```

Comment:
I suggest adding sample_name="SAMPLE" as an input and using it instead of a heardcoded string